doc/sphinx/conf.py

```diff
@@ -28,6 +28,7 @@
     'sphinx.ext.autodoc',
     'sphinx.ext.intersphinx',
     'sphinx_copybutton',
+    'sphinx_tabs.tabs',
```

Comment:
Instead of the `sphinx-tabs` library, let's use [`sphinx-design`](https://github.com/executablebooks/sphinx-design). Mainly the tabs look better and they are specifically designed to work with our theme, `furo`. It should not be fairly straightforward to convert. I was able to update one of the samples below to the new extension format and overall looked much cleaner in the rendered HTML:
```rst
.. tab-set::

    .. tab-item:: ``BlobDataset``
    
        .. literalinclude:: ../../samples/blob_dataset_sample.py
            :lines: 16, 20, 39-45
    
    .. tab-item:: ``IterableBlobDataset``
    
        .. literalinclude:: ../../samples/iterable_blob_dataset_sample.py 
            :lines: 16, 20, 39-45
```